### PR TITLE
Remove the `Serial` constraint from ListApp repo test

### PR DESCRIPTION
**Pull request by danail-branekov**
## Is there a related GitHub Issue?
https://github.com/cloudfoundry/cf-k8s-controllers/issues/303

## What is this change about?
Get rid of `Serial` in list app repository tests
* The app repo uses a restricted user client to list apps, therefore
  it can only return apps in namespaces the user client is permissioned
  to. Therefore it makes no sense to ensure that there are no apps in
  other spaces (as the repo does not "see" them at all)
* While being here introduced misc improvements:
  * Moved `cfapp2`, `cfApp3`, `space2`, `space3` down to contexts that
    really need them (i.e. they are not created for every test)
  * Extracted some invocations of the function under test in
    `JustBeforeEach` to delete a bunch of code repetition

## Does this PR introduce a breaking change?
No

## Acceptance Steps
Green tests

### CODE REVIEW

File: api/repositories/app_repository_test.go [hunk 1]
```diff
@@ -183,214 +178,178 @@ var _ = Describe("AppRepository", func() {
...
-						MatchFields(IgnoreExtras, Fields{"GUID": Equal(cfApp2.Name)}),
-					))
+		When("there are apps in non-cf namespaces", func() {
+			var nonCFApp *workloadsv1alpha1.CFApp
 
-					sortedByName := sort.SliceIsSorted(appList, func(i, j int) bool {
-						return appList[i].Name < appList[j].Name
-					})
+			BeforeEach(func() {
+				nonCFNamespace := prefixedGUID("non-cf")
```
Comment (georgethebeatle) [inline api/repositories/app_repository_test.go:223]: The non-cf namespace needs to be cleaned up in an AfterEach

File: api/repositories/app_repository_test.go [hunk 2]
```diff
@@ -135,16 +130,15 @@ var _ = Describe("AppRepository", func() {
...
 
 		BeforeEach(func() {
-			cfApp1 = createApp(space1.Name)
-			querySpaceName = space1.Name
+			querySpaceName = space.Name
 		})
 
 		AfterEach(func() {
-			Expect(k8sClient.Delete(context.Background(), cfApp1)).To(Succeed())
+			Expect(k8sClient.Delete(context.Background(), cfApp)).To(Succeed())
```
Comment (georgethebeatle) [inline api/repositories/app_repository_test.go:137]: Is this AfterEach needed if we are going to delete the space anyway?

File: api/repositories/app_repository_test.go [hunk 3]
```diff
@@ -66,28 +65,28 @@ var _ = Describe("AppRepository", func() {
 
 		When("authorized in the space", func() {
 			BeforeEach(func() {
-				createRoleBinding(testCtx, userName, spaceDeveloperRole.Name, space1.Name)
+				createRoleBinding(testCtx, userName, spaceDeveloperRole.Name, space.Name)
 			})
 
 			AfterEach(func() {
-				Expect(k8sClient.Delete(context.Background(), cfApp1)).To(Succeed())
+				Expect(k8sClient.Delete(context.Background(), cfApp)).To(Succeed())
```
Comment (georgethebeatle) [inline api/repositories/app_repository_test.go:72]: Is this AfterEach needed if we are going to delete the space anyway?

**Review by georgethebeatle — APPROVED** [review]
LGTM with some minor comments

**Comment by georgethebeatle** [comment]
Opps, this auto-merged. If you feel like my comments make any sense open a new PR and I will review it, should be a quickie.

File: api/repositories/app_repository_test.go (hunk 1 above)
Comment (danail-branekov) [reply api/repositories/app_repository_test.go:223]: Why? Env tests generally do not need clean up as their environment is thrown away anyway. How about getting rid of all cleanup?

File: api/repositories/app_repository_test.go (hunk 2 above)
Comment (danail-branekov) [reply api/repositories/app_repository_test.go:137]: Deleting spaces in env tests does not delete the resources they contain. However, as per my previous comment, cleanup in env tests is not really needed.

**Pull request merged**